Tests/TensorFlowTests/LayerTests.swift

```diff
@@ -205,6 +205,23 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output.shape, expected)
     }
 
+    func testEmbedding() {       
+        var layer = Embedding<Float>(vocabularySize: 3, embeddingSize: 5)       
+        var data = Tensor<Int32>(shape: [2,3], scalars: [0, 1, 2, 1, 2, 2])
+        var input = Embedding<Float>.Input(indices: data)
```

Comment:
This line shows that it's suboptimal to define `Input` as a nested struct inside `Embedding` because it requires a generic parameter (i.e. `Embedding.Scalar`) to create an `Input`. As such, I think it's better to define a top-level struct `EmbeddingInput`.

```swift
@frozen
public struct EmbeddingInput: Differentiable {
    @noDerivative public var indices: Tensor<Int32>
}
```

This way you can create an embedding input directly: `Embedding(indices: data)` and not care about the output scalar type of the embedding layer.